src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts

```diff
@@ -442,38 +456,153 @@ export class InstructorCoursesPageComponent implements OnInit {
-    return this.feedbackSessionsService
-      .createFeedbackSession(newCourseId, this.toFbSessionCreationReqWithName(fromFeedbackSession, oldCourseId));
+  private copyFeedbackSession(fromFeedbackSession: FeedbackSession, newCourseId: string,
+                              newTimeZone: string, oldCourseId: string): Observable<FeedbackSession> {
+    return this.feedbackSessionsService.createFeedbackSession(newCourseId,
+        this.toFbSessionCreationReqWithName(fromFeedbackSession, newTimeZone, oldCourseId));
   }
 
   /**
    * Creates a FeedbackSessionCreateRequest with the provided name.
    */
-  private toFbSessionCreationReqWithName(fromFeedbackSession: FeedbackSession, oldCourseId: string):
-      FeedbackSessionCreateRequest {
+  private toFbSessionCreationReqWithName(fromFeedbackSession: FeedbackSession, newTimeZone: string,
+                                         oldCourseId: string): FeedbackSessionCreateRequest {
+    // Local constants
+    const twoHoursBeforeNow = moment().tz(newTimeZone).subtract(2, 'hours')
+        .valueOf();
+    const twoDaysFromNowRoundedUp = moment().tz(newTimeZone).add(2, 'days').startOf('hour')
+        .valueOf();
+    const sevenDaysFromNowRoundedUp = moment().tz(newTimeZone).add(7, 'days').startOf('hour')
+        .valueOf();
+    const ninetyDaysFromNow = moment().tz(newTimeZone).add(90, 'days')
+        .valueOf();
+    const ninetyDaysFromNowRoundedUp = moment().tz(newTimeZone).add(90, 'days').startOf('hour')
+        .valueOf();
+    const oneHundredAndEightyDaysFromNow = moment().tz(newTimeZone).add(180, 'days')
+        .valueOf();
+    const oneHundredAndEightyDaysFromNowRoundedUp = moment().tz(newTimeZone).add(180, 'days')
+        .startOf('hour')
+        .valueOf();
+
+    // Preprocess timestamps to adhere to feedback session timestamps constraints
+    let isModified = false;
+
+    let copiedSubmissionStartTimestamp = fromFeedbackSession.submissionStartTimestamp;
+    if (copiedSubmissionStartTimestamp < twoHoursBeforeNow) {
+      copiedSubmissionStartTimestamp = twoDaysFromNowRoundedUp;
+      isModified = true;
+    } else if (copiedSubmissionStartTimestamp > ninetyDaysFromNow) {
+      copiedSubmissionStartTimestamp = ninetyDaysFromNowRoundedUp;
+      isModified = true;
+    }
+
+    let copiedSubmissionEndTimestamp = fromFeedbackSession.submissionEndTimestamp;
+    if (copiedSubmissionEndTimestamp < copiedSubmissionStartTimestamp) {
+      copiedSubmissionEndTimestamp = sevenDaysFromNowRoundedUp;
+      isModified = true;
+    } else if (copiedSubmissionEndTimestamp > oneHundredAndEightyDaysFromNow) {
+      copiedSubmissionEndTimestamp = oneHundredAndEightyDaysFromNowRoundedUp;
+      isModified = true;
+    }
+
+    let copiedSessionVisibleSetting = fromFeedbackSession.sessionVisibleSetting;
+    let copiedCustomSessionVisibleTimestamp = fromFeedbackSession.customSessionVisibleTimestamp!;
+    const thirtyDaysFromSubmissionStart = moment(copiedSubmissionStartTimestamp)
```

Comment:
Variable name should be `thirtyDaysBeforeSubmissionStart`, similarly for the rounded up version